test/integration/dryrun/dryrun_test.go

```diff
@@ -0,0 +1,619 @@
+			kind := gvk.Kind
+
+			mapping := &meta.RESTMapping{
+				Resource:         resourceToPersist.gvr,
+				GroupVersionKind: resourceToPersist.gvk,
+				Scope:            meta.RESTScopeRoot,
+			}
+			if resourceToPersist.namespaced {
+				mapping.Scope = meta.RESTScopeNamespace
+			}
+
+			if kindWhiteList.Has(kind) {
+				t.Skip("whitelisted")
+			}
+
+			testData, hasTest := dryrunData[gvResource]
+
+			if !hasTest {
+				t.Fatalf("no test data for %s.  Please add a test for your new type to dryrunData.", gvResource)
+			}
+
+			// we don't require GVK on the data we provide, so we fill it in here.  We could, but that seems extraneous.
+			typeMetaAdder := map[string]interface{}{}
+			err := json.Unmarshal([]byte(testData.stub), &typeMetaAdder)
+			if err != nil {
+				t.Fatalf("failed to unmarshal stub (%v): %v", testData.stub, err)
+			}
+			typeMetaAdder["apiVersion"] = mapping.GroupVersionKind.GroupVersion().String()
+			typeMetaAdder["kind"] = mapping.GroupVersionKind.Kind
+
+			rsc := dClient.Resource(mapping.Resource).Namespace(testNamespace)
+			if mapping.Scope == meta.RESTScopeRoot {
+				rsc = dClient.Resource(mapping.Resource)
+			}
+			obj := &unstructured.Unstructured{Object: typeMetaAdder}
+			if _, err := rsc.Create(obj, metav1.CreateOptions{DryRun: []string{metav1.DryRunAll}}); err != nil {
+				t.Fatalf("failed to dry-run create stub for %s: %#v", gvResource, err)
+			}
+
+			if _, err := rsc.Get(obj.GetName(), metav1.GetOptions{}); err == nil {
+				t.Fatalf("object shouldn't exist")
+			}
+
+			if _, err := rsc.Create(obj, metav1.CreateOptions{}); err != nil {
+				t.Fatalf("failed to create stub for %s: %#v", gvResource, err)
+			}
+
+			obj, err = rsc.Get(obj.GetName(), metav1.GetOptions{})
+			if err != nil {
+				t.Fatalf("failed to retrieve object: %v", err)
+			}
+
+			if !testData.skipUpdate {
+				obj.SetAnnotations(map[string]string{"update": "true"})
+				obj, err = rsc.Update(obj, metav1.UpdateOptions{DryRun: []string{metav1.DryRunAll}})
+				if err != nil {
+					t.Fatalf("failed to dry-run update object: %v", err)
+				}
+				if _, ok := obj.GetAnnotations()["update"]; !ok {
+					t.Fatalf("dry-run updated annotations should be returned, got: %v", obj.GetAnnotations())
+				}
+			}
+			patch := []byte(`{"metadata":{"annotations":{"patch": "true"}}}`)
+			obj, err = rsc.Patch(obj.GetName(), types.MergePatchType, patch, metav1.UpdateOptions{DryRun: []string{metav1.DryRunAll}})
+			if err != nil {
+				t.Fatalf("failed to dry-run patch object: %v", err)
+			}
+			if _, ok := obj.GetAnnotations()["patch"]; !ok {
+				t.Fatalf("dry-run patched annotations should be returned, got: %v", obj.GetAnnotations())
+			}
+
+			if obj, err := rsc.Get(obj.GetName(), metav1.GetOptions{}, "status"); err == nil {
+				_, err = rsc.Update(obj, metav1.UpdateOptions{DryRun: []string{metav1.DryRunAll}})
+				if err != nil {
+					t.Fatalf("dry-run update on status resource failed: %v", err)
+				}
```

Comment:
Verify that the object wasn't actually changed, and that the return value was (you'll have to tweak something in obj, I think annotations should work).